lib/apiservers/engine/backends/container.go

```diff
@@ -481,8 +406,11 @@ func (c *Container) ContainerExecStart(ctx context.Context, eid string, stdin io
 		defer cancel()
 
 		// we do not return an error here if this fails. TODO: Investigate what exactly happens on error here...
+		// FIXME: This being in a retry could lead to multiple writes to stdout(?)
```

Comment:
This should not be writing an error in the goroutine - this needs properly integrating with the retry logic and context cancellation.

I'm unsure why this is in a goroutine actually. We should check to see if an `exec_start` event is generated when trying to exec an non-existent binary in regular docker - if it reflects the container behaviour it will not generate that event on failure.

Please figure out if this needs to be in a goroutine and comment why in the code or adjust it appropriately.